### PR TITLE
Resubscribe single optin v2

### PR DESCRIPTION
**Pull request by kpu23**

### CODE REVIEW

File: public/partials/shortcodes/process/process_form_submission.php [hunk 1]
```diff
@@ -167,20 +167,20 @@
 
 // If this member does not exist, then we need to add the status_if_new flag and set our $new_subscriber variable
 // Likewise, if this member exists but their status is 'pending' it means we're dealing with a double opt-in list and they never confirmed
-// Or, if this member but their status is 'unsubscribed' it means we're dealing with someone who unsubscribed and they need to re-subscribe
+// Or, if this member exists but their status is 'unsubscribed' it means we're dealing with someone who unsubscribed and they need to re-subscribe (this could be single opt-in or double opt-in)
 // Continue as if they're a new member to force another double opt-in email
-$double_optin_resubscribe = is_array( $member_exists ) && isset( $member_exists['status'] ) && ( $member_exists['status'] === 'pending' || $member_exists['status'] === 'unsubscribed' ) ? true : false;
+$was_subscribed = is_array( $member_exists ) && isset( $member_exists['status'] ) && ( $member_exists['status'] === 'pending' || $member_exists['status'] === 'unsubscribed' ) ? true : false;
```
Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission.php]: For this line, you can drop `is_array( $member_exists )`, and you can also drop the `? true : false` from the end. The various comparison operators (`&&`, `||`, etc) return a boolean value natively.

File: public/partials/shortcodes/process/process_form_submission.php [hunk 2]
```diff
@@ -167,20 +167,20 @@
...
 
-if ( is_wp_error( $member_exists ) || $double_optin_resubscribe === true ) {
+if ( is_wp_error( $member_exists ) || $was_subscribed === true ) {
```
Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission.php]: Yoda conditions, WordPress likes. Put the `true` before `$was_subscribed`, you should.

File: public/partials/shortcodes/process/process_form_submission_ajax.php [hunk 3]
```diff
@@ -129,21 +129,21 @@
...
 // Likewise, if this member exists but their status is 'pending' it means we're dealing with a double opt-in list and they never confirmed
-// Or, if this member but their status is 'unsubscribed' it means we're dealing with someone who unsubscribed and they need to re-subscribe
+// Or, if this member exists but their status is 'unsubscribed' it means we're dealing with someone who unsubscribed and they need to re-subscribe (this could be single opt-in or double opt-in)
 // Continue as if they're a new member to force another double opt-in email
-$double_optin_resubscribe = is_array( $member_exists ) && isset( $member_exists['status'] ) && ( $member_exists['status'] === 'pending' || $member_exists['status'] === 'unsubscribed' ) ? true : false;
+$was_subscribed = is_array( $member_exists ) && isset( $member_exists['status'] ) && ( $member_exists['status'] === 'pending' || $member_exists['status'] === 'unsubscribed' ) ? true : false;
 
-if ( is_wp_error( $member_exists ) || $double_optin_resubscribe === true ) {
+if ( is_wp_error( $member_exists ) || $was_subscribed === true ) {
```
Comment (JPry) [inline public/partials/shortcodes/process/process_form_submission_ajax.php]: The same two items from the other file also apply here.

File: public/partials/shortcodes/process/process_form_submission.php (hunk 2 above)
Comment (kpu23) [reply public/partials/shortcodes/process/process_form_submission.php]: I'm sorry I just can't get on board with Yoda conditions. In English it's confusing to read the sentence `Yoda conditions, WordPress likes.` Likewise in code it's confusing to read `if ( true === $was_subscribed )`

File: public/partials/shortcodes/process/process_form_submission.php (hunk 1 above)
Comment (kpu23) [reply public/partials/shortcodes/process/process_form_submission.php]: Good point. Thank you.

**Pull request closed without merging**